examples/keycloak/README.md

```diff
@@ -0,0 +1,270 @@
+kind: Ingress
+metadata:
+  name: keycloak-ingress
+spec:
+  rules:
+    # provide the actual Ingress for `host` value:
+    # use the following command to get the subdomain:
+    #    ibmcloud ks cluster get --cluster <cluster-name> | grep Ingress
+    # any prefix can be defined as a result (e.g.):
+    # - host: tsi-keycloak-v001.tsi-fra02-5240a746a818fd9d58aa25c34ecfe-0000.eu-de.containers.appdomain.cloud
+    # provide the actual Ingress for `host` value:
+  - host: tsi-keycloak.my-tsi-cluster-8abee0d19746a818fd9d58aa25c34ecfe-0000.eu-de.containers.appdomain.cloud
+    http:
+      paths:
+      - backend:
+          serviceName: tsi-keycloak
+          servicePort: 9090
+        path: /
+```
+
+create ingress:
+```console
+$ kubectl -n tsi-keycloak create -f ingress-IKS.yaml
+```
+</details>
+
+To access Keycloak on OpenShift (including IKS ROKS)
+<details><summary>Click to view OpenShift steps</summary>
+
+This assumes the OpenShift command line is already installed. Otherwise see
+the [documentation](https://docs.openshift.com/container-platform/4.2/cli_reference/openshift_cli/getting-started-cli.html)
+and you can get `oc` cli from https://mirror.openshift.com/pub/openshift-v4/clients/oc/4.3/
+
+```console
+oc -n tsi-keycloak expose svc/tsi-keycloak
+export KEYCLOAK_ADDR="http://$(oc -n tsi-keycloak get route tsi-keycloak -o jsonpath='{.spec.host}')"
+```
+
+</details>
+
+Keycloak Configuration steps:
+* Using the Keycloak URL obtained above, connect to the server with a browser and
+login using admin userid and password, as specified in the [/examples/keycloak/keycloak.yaml](/examples/keycloak/keycloak.yaml)
+* In upper left corner, below Keycloak logo, click on `Master` and select `Add realm`
+* Select `Import` and navigate to [/examples/keycloak/tsi-realm-export.json](/examples/keycloak/tsi-realm-export.json) file. Press Create.
+
+This should instantiate a new `Tsi-realm` realm that we would use for the demo.
+
+## Keycloak Access Policies
+By loading the `tsi-realm-export.json` into Keycloak, we configured most of components
+needed for the demo. Now we just need to modify the policies to match your
+environment.
+
+During the [installation of TSI](/README.md#install-trusted-service-identity-framework), we used `CLUSTER_NAME` and `REGION` attributes. They were either given to us by the
+service provider (IBM Cloud IKS or ROKS), or we defined them ourselves manually
+(e.g. in Minikube)
+
+These values, along with `images`, `namespace`,`pod-id`, etc, represent the identity
+of the workload. We can use them to define policies that would restrict access
+to specific resources.
+
+For example, let's take a look a the Client `tsi-client`:
+* Select `Clients`  --> `tsi-client`
+* Select `Authorization` tab
+* Select `Policies` tab
+* Take a look at the given policies.
+  - `tsi-policy` grants access if the `region` contains `eu`
+  - `strict-image-policy` grants access if the `image-names` contains `myubuntu`
```

Comment:
I think in this verison of the realm it is 

`image-policy-only-nginx` and only for nginx images, i looked at the realm.json file.

Reply:
Actually, it is `strict-image-policy`:

```
          {
            "id": "174cf084-4c24-45e4-b158-32ad0a87ae80",
            "name": "strict-image-policy",
            "description": "Enforce image policy of workload",
            "type": "js",
            "logic": "POSITIVE",
            "decisionStrategy": "UNANIMOUS",
            "config": {
              "code": "var context = $evaluation.getContext();\nvar identity = context.getIdentity();\nvar attributes = identity.getAttributes();\n\nif (attributes.exists(\"image-names\")) {\n    var imageNames = attributes.getValue(\"image-names\").asString(0);\n    if (imageNames.contains(\"myubuntu\")) {\n            $evaluation.grant();\n    }\n}"
            }
          },
```